txtorcon/endpoints.py

```diff
@@ -667,6 +668,7 @@ class TorClientEndpoint(object):
     socks_ports_to_try = [9050, 9150]
 
     def __init__(self, host, port,
+                 socks_socket=None,
                  socks_hostname=None, socks_port=None,
```

Comment:
It seems like it'd be better to have just one way to specify "how to connect to SOCKS" -- right now there's _proxy_endpoint_generator= and socks_socket= (and socks_hostname/socks_port) -- can we just have e.g. "socks_endpoint=None" and if it's None, we try some default things? We could change the "list of default things" to be actual endpoint-strings, too, so that we could add "unix:/var/log/tor/control" to it as well. In any case, I think it'd be better if "socks_socket" were at least "socks_endpoint" so that a user can manually specify any sort of endpoint (including of course UNIXClientEndpoint).

Backwards-compatibilitiy: socks_hostname/socks_port are public API; we can capture **kwargs and do something sensible if these were specified (i.e. pretend they said socks_endpoint=TCP4ClientEndpoint(host, port)).

I see I dropped _proxy_endpoint_generator from "the new stuff" so maybe we can accomplish that here, too?